shared/js/pageStore.js

```diff
@@ -100,6 +115,72 @@ loop.store.PageStore = function(mozL10n) {
           label: mozL10n.get("snackbar_page_deleted")
         }));
       }
+    },
+
+    /**
+     * Check whether this store already contains this URL.  Note that
+     * we currently use an exact string compare for this; we might want
+     * to consider switching to something less strict if it makes sense
+     * from a product standpoint.
+     *
+     * @param {String} url - the thing we're looking for
+     * @return {Boolean}
+     */
+    _hasPageForUrl(url) {
+      let pages = this._storeState.pages;
+      pages = pages.filter(page =>
+        page.url === url
```

Comment:
Please add a XXX akita comment to keep in mind this: `we might want to consider switching to something less strict if it makes sense from a product standpoint.`